app/controllers/verify/finance_controller.rb

```diff
@@ -9,7 +9,10 @@ def new
     end
 
     def create
-      if step.complete
+      result = step.submit
```

Comment:
I understand that you're trying to get consistency in the controller between how a `Step` works and how a `Form` works.

Still, calling a method `submit` on a `Step` feels a little awkward, grammatically. I submit a form. I complete a step. It smells a little like false equivalence. OTOH, naming things is the Hardest Problem, and this isn't a blocker for me.